### PR TITLE
[Program:GCI] Added new API endpoint-count pending mentorship requests

### LINKED ISSUE #320: Create an API Endpoint to Count Pending Mentorship Relations
**Describe the solution you'd like**
Create a new API Endpoint at GET/mentorship_relations/count_pending which counts the number of pending mentorship relations of the current user. Return a dictionary response with key "count" and value as the number of pending mentorship relations.

**Additional context**
This issue has been created for students participating in GCI'19.

### PR DESCRIPTION
**Pull request by rudrakshjhalanii07**
### Description 
Created a new API endpoint to count the number of pending mentorship requests of the current user.

Fixes #320 

### Type of Change:
**Delete irrelevant options.**

- Code
- Quality Assurance
- Documentation

**Code/Quality Assurance Only**
- This change requires a documentation update (software upgrade on readme file)
- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
This has been tested successfully by me.

`Screenshot #1`
IMG_20191229_091136
PicsArt_12-29-01 10 08

`Screenshot #2`
IMG_20191229_091232
IMG_20191229_130224

### Checklist:
**Delete irrelevant options.**

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] Any dependent changes have been merged
- [ ] Update Postman API at /docs folder
- [x] Update Swagger documentation and the exported file at /docs folder
- [ ] Update requirements.txt

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

File: app/api/dao/mentorship_relation.py [hunk 1]
```diff
@@ -365,3 +365,27 @@ def list_pending_mentorship_relations(user_id):
...
 
         return pending_requests, 200
+
+    @staticmethod
+    @email_verification_required
+    def count_pending_mentorship_relations(user_id):
+        """Counts all the pending mentorship requests of the user.
+
+        Args:
+            user_id: ID of the user counting all his pending mentorship requests.
```
Comment (annabauza) [inline app/api/dao/mentorship_relation.py]: We don't know user gender.

File: app/api/dao/mentorship_relation.py [hunk 2]
```diff
@@ -365,3 +365,27 @@ def list_pending_mentorship_relations(user_id):
...
+
+        Returns:
+            message: A message containing the number of pending mentorship requests of the user along with the success status code,
+                     or failure status code if otherwise.
```
Comment (annabauza) [inline app/api/dao/mentorship_relation.py]: `success status code, or failure status code` - does it really return any status here?

File: app/api/dao/mentorship_relation.py [hunk 3]
```diff
@@ -365,3 +365,27 @@ def list_pending_mentorship_relations(user_id):
...
+        """
+
+        user = UserModel.find_by_id(user_id)
+        now_timestamp = datetime.now().timestamp()
+        all_relations = user.mentor_relations + user.mentee_relations
+        pending_request_count = 0
+
+        for relation in all_relations:
```
Comment (annabauza) [inline app/api/dao/mentorship_relation.py]: Would be nicer to use lambda here.

File: docs/swagger.json [hunk 4]
```diff
@@ -522,6 +522,39 @@
                 ]
             }
         },
+        "/mentorship_relations/count_pending": {
```
Comment (annabauza) [inline docs/swagger.json:525]: I don't think we should modify swagger manually but will confirm that with other mentors

Comment (annabauza) [inline docs/swagger.json:525]: I don't think we should modify swagger manually but will confirm that with other mentors

**Review by annabauza — CHANGES_REQUESTED** [review]
Hi.

I have left you some comments. :)

File: app/api/dao/mentorship_relation.py (hunk 3 above)
Comment (rudrakshjhalanii07) [reply app/api/dao/mentorship_relation.py]: Gotta learn that, will update soon!

File: docs/swagger.json (hunk 4 above)
Comment (rudrakshjhalanii07) [reply docs/swagger.json:525]: Ok, please let me know after confirmation. :)

File: app/api/dao/mentorship_relation.py (hunk 3 above)
Comment (rudrakshjhalanii07) [reply app/api/dao/mentorship_relation.py]: Updated!

**Comment by rudrakshjhalanii07** [comment]
I have updated all the asked changes.

File: docs/swagger.json (hunk 4 above)
Comment (annabauza) [reply docs/swagger.json:525]: hey the mentors didn't come back yet but I don't want to block you with the task, so let's just leave it here for now :)

**Review by annabauza — APPROVED** [review]

File: docs/swagger.json (hunk 4 above)
Comment (annabauza) [reply docs/swagger.json:525]: just received confirmation - we edit the code and the documentation is autogenerated.

Comment (annabauza) [reply docs/swagger.json:525]: just received confirmation - we edit the code and the documentation is autogenerated.

**Comment by isabelcosta** [comment]
Thank you for your contribution!
I will close this PR, because this is a not a desirable feature for the app.

**Pull request closed without merging**